### PR TITLE
refactor: refactor logger

### PR DESCRIPTION
**Pull request by rockmagma02**
## Description

refactor(logger): refactor logger

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://omnisafe.readthedocs.io/en/latest/developer/contributing.html) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: omnisafe/algorithms/on_policy/base/policy_gradient.py [hunk 1]
```diff
@@ -135,6 +143,56 @@ def __init__(self, env_id: str, cfgs: Config) -> None:
...
 
+        self._init_log()
+
+    def _init_log(self):
+        self.logger.register_key('Train/Epoch')
+        self.logger.register_key('Metrics/EpRet')
+        self.logger.register_key('Metrics/EpCost')
+        self.logger.register_key('Metrics/EpLen')
+
+        # # log information about actor
```
Comment (Gaiejj) [inline omnisafe/algorithms/on_policy/base/policy_gradient.py:154]: why double #

File: omnisafe/algorithms/on_policy/base/policy_gradient.py [hunk 2]
```diff
@@ -81,18 +81,26 @@ def __init__(self, env_id: str, cfgs: Config) -> None:
...
-        # set up experience buffer
 
+        # set up logger and save configuration to disk
+        self.logger = Logger(
+            output_dir=cfgs.data_dir,
+            exp_name=cfgs.exp_name,
+            seed=cfgs.seed,
+            use_tensorboard=cfgs.use_tensorboard,
+            use_wandb=cfgs.use_wandb,
+            config=cfgs,
```
Comment (Gaiejj) [inline omnisafe/algorithms/on_policy/base/policy_gradient.py:99]: if `configs=cfgs`, why specifically passing `seed=cfgs.seed`

**Review by Gaiejj — CHANGES_REQUESTED** [review]

File: omnisafe/algorithms/on_policy/base/policy_gradient.py (hunk 1 above)
Comment (rockmagma02) [reply omnisafe/algorithms/on_policy/base/policy_gradient.py:154]: typo bugs, I will fix right now

File: omnisafe/algorithms/on_policy/base/policy_gradient.py (hunk 2 above)
Comment (rockmagma02) [reply omnisafe/algorithms/on_policy/base/policy_gradient.py:99]: The argument config is passed to the Logger for the purpose of saving to Weights and Biases. However, if the user does not require this functionality, they may pass None as the config argument. In this sitution, user can only pass `seed`, `output_dir` etc for basic logger use. This provides greater flexibility and eliminates unnecessary dependencies. To accommodate varying needs, another argument can be utilized instead, making the code more modular, adaptable, reusable, and maintainable.

**Review by zmsn-2077 — APPROVED** [review]
LGTM.

**Review by Gaiejj — APPROVED** [review]
LGTM.

**Pull request merged**